tests/group_functions/group_functions_common.h

```diff
@@ -0,0 +1,305 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2023 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include "type_coverage.h"
+
+#include <catch2/catch_template_test_macros.hpp>
+
+/*
+ * FIXME: hipSYCL does not implement size member function of sycl::vec
+ * As a result the following implementation is not working
+ * Workaround is presented below
+
+template<typename T, typename U>
+bool equal(const T& a, const U& b)
+{
+  return value_operations::are_equal(a, b);
+}
+*/
+template <typename T, typename U>
+bool equal_impl(const T& a, const U& b) {
+  return a == b;
+}
+
+template <typename T, int N>
+bool equal_impl(const sycl::vec<T, N>& a, const sycl::vec<T, N>& b) {
+  bool res = true;
+  for (int i = 0; i < N; ++i) res &= (a[i] == b[i]);
+  return res;
+}
+
+template <typename T, int N, typename U>
+bool equal_impl(const sycl::vec<T, N>& a, const U& b) {
+  bool res = true;
+  for (int i = 0; i < N; ++i) res &= (a[i] == b);
+  return res;
+}
+
+template <typename T, typename U>
+bool equal(const T& a, const U& b) {
+  return equal_impl(a, b);
+}
+
+/*
+ * FIXME: hipSYCL cannot construct vector from 1 value
+ * As a result the helper below is needed
+ */
+template <typename T, typename U>
+T init_helper(const U init) {
```

Comment:
Use a name in relation with the semantics and the comment.

Reply:
I'm stuck here, please suggest one :)

Reply:
Something with `splat` like http://p12tic.github.io/libsimdpp/v2.0~rc1/libsimdpp/misc/splat.html, https://reviews.llvm.org/D65286, https://github.com/llvm/llvm-project/issues/56658 ?
`splat_init::assign` then?

Reply:
I changed the naming to `splat_init` in the latest commit.